rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -190,7 +190,20 @@ class Player
     Arguments arguments({"--ros-args", "--log-level", log_level});
     rclcpp::init(arguments.argc(), arguments.argv());
     player_ = std::make_shared<rosbag2_transport::Player>(storage_options, play_options, node_name);
+  }
+
+  virtual ~Player()
+  {
+    stop_spin();
+    rclcpp::shutdown();
+  }
 
+  void start_spin()
+  {
+    if (!player_) {
+      throw std::runtime_error("Player is not initialized. Please use constructor with "
+        "storage and play options.");
+    }
```

Comment:
Need to avoid calling `start_spin` twice if it is already spinning.

Reply:
see e1fc20bd91a694a73fcb5cdbbeb4976f1d59fedc